bndtools.core/src/org/bndtools/core/editors/ImportPackageQuickFixProcessor.java

```diff
@@ -0,0 +1,578 @@
+                try {
+                    ICON = Icons.desc("bundle")
+                        .createImage();
+                } catch (RuntimeException e) {
+                    logger.logError("Couldn't load bundle image", e);
+                }
+            }
+            return ICON;
+        }
+
+        @Override
+        public IContextInformation getContextInformation() {
+            return new IContextInformation() {
+
+                @Override
+                public String getContextDisplayString() {
+                    return "Added " + fBundle + " to build path";
+                }
+
+                @Override
+                public Image getImage() {
+                    return null;
+                }
+
+                @Override
+                public String getInformationDisplayString() {
+                    return "Added " + fBundle + " to build path - info";
+                }
+
+            };
+        }
+
+        @Override
+        public int getRelevance() {
+            return fRelevance;
+        }
+    }
+
+    // Wrap these methods to make it easier to test.
+    Workspace getWorkspace() throws Exception {
+        return Central.getWorkspace();
+    }
+
+    // Wrap these methods to make it easier to test.
+    Repository getWorkspaceRepo() throws Exception {
+        return Central.getWorkspaceR5Repository();
+    }
+
+    List<RepositoryPlugin> listRepositories() {
+        return RepositoryUtils.listRepositories(true);
+    }
+
+    IJavaCompletionProposal[] getSuggestions(Set<String> pkgs, IInvocationContext context) {
+        List<RepositoryPlugin> ps = listRepositories();
+
+        final BndBuildPathHandler handler = getBuildPathHandler(context);
+        SetMultimap<String, String> bundles = MultimapBuilder.treeKeys()
+            .treeSetValues()
+            .build();
+        String wsName = null;
+        for (String pkg : pkgs) {
+            for (RepositoryPlugin p : ps) {
+                Set<Capability> caps = null;
+                if (p instanceof Repository) {
+                    caps = searchRepository((Repository) p, pkg);
+                } else if (p instanceof WorkspaceRepository) {
+                    try {
+                        caps = searchRepository(getWorkspaceRepo(), pkg);
+                        wsName = p.getName();
+                    } catch (Exception e) {
+                        logger.logError("Error trying to fetch the repository for the current workspace", e);
+                    }
+                }
+                if (caps == null) {
+                    continue;
+                }
+                for (Capability cap : caps) {
+                    final String bsn = capabilityToBSN(cap);
+                    try {
+                        if (!handler.containsBundle(bsn)) {
+                            bundles.put(bsn, p.getName());
+                        }
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
This is not useful in an IDE since no one may see it. Better to log.

Reply:
Yes, that was an oversight. I have removed the try/catch and allowed the `CoreException` to propagate up to the Java core through `getCorrections()`. I have verified that Eclipse takes care of reporting the error (to its error log).